keygen/CMakeLists.txt

```diff
@@ -0,0 +1,62 @@
+cmake_minimum_required(VERSION 3.1 FATAL_ERROR)
+
+project(enftun-keygen
+  VERSION 0.0.1
+  LANGUAGES C
+)
+
+list(APPEND CMAKE_MODULE_PATH "${enftun-keygen_SOURCE_DIR}/cmake")
+
+include(GNUInstallDirs)
+
+find_package(OpenSSL REQUIRED QUIET 1.1)
+find_package(Jansson REQUIRED)
+find_package(CURL REQUIRED)
+if (NOT TARGET sodium)
```

Comment:
libsodium is used for generating the random serial number for the x.509 certs. I figured just initializing the libsodium library and calling their random_bytes_buf function is easier than dealing with seeding the OpenSSL rng, since we already require that library, as you said.

Reply:
I was wrong, the OpenSSL rng does automatically seed itself, so we could use that instead and not have to fiddle with seeding it (at least, I feel comfortable with the corner cases for this use-case).

It isn't critical to use cryptographically-strong random bytes for the serial number (this is a self-signed cert, so any attacker can do whatever they want outside of this project). I just didn't want to use the same value for every cert, or use `rand`, or have to fiddle with seeding.

It doesn't seed itself if `/dev/urandom` isn't available (and, apparently, it can use CryptGenRandom on Windows), but that's probably the same limitation as libsodium. I don't know if it fails silently or what, but again this isn't that sensitive so it's probably OK.

You've gotta love OpenSSL documentation: https://wiki.openssl.org/index.php/Random_Numbers#Seeds
"Even if the docs state a seed is not needed, you should seed it anyway."
... a few lines later ...
"OpenSSL will attempt to seed the random number generator automatically upon instantiation"
Also, "Be careful when deferring to RAND_poll on some Unix systems because it does not seed the generator". So, we'll automatically seed it for, except for when we won't.
And "(the seeding can have) negative interactions on newer Windows platforms, so your program could hang or crash"

I don't know if we need to be concerned about this for our other use of OpenSSL (actually generating the keys and signatures, running the TLS session in the main library, etc.). Probably. Ugh.